### ISSUE TITLE
This image causes datamatrix decode to hard crash

### ISSUE THREAD

**Issue by raylutz** [NaturalHistoryMuseum/pylibdmtx#62]
I found I had to wrap the pylibdmtx in an external subprocess (i.e. copy the source image to a file and then have a CLI envoked script that calls pylibdmtx and writes the result to a file) because this image causes everything to just hard stop, with no error messages or exceptions raised. I always treasure this sort of test data because it should be fairly easy for someone familiar with the code to fix the problem, but it isn't something I have the knowledge to do. 

![datamatrix_temp_09992_00008_000024_0](https://user-images.githubusercontent.com/14955977/115151516-064d6600-a022-11eb-8f13-af5804935d7a.png)

Enjoy!

**Comment by tusharj333** [comment]
Hi, @raylutz  Did you find any fix for this issue?

**Comment by raylutz** [comment]
> Hi, @raylutz Did you find any fix for this issue?

No, the datamatrix code still crashes, but by wrapping it in a subprocess, it will not cause the calling program to crash. Try/Except block does not work in this example.

**Comment by MathijsNL** [comment]
Interesting find.

Using the library (dmtxread) also crashes, so the problem lies in the lib and not in the wrapper.


```
+++ exited with 40 +++
```

Adding a 10 pixel white border seems to fix the issue. Maybe someone more experienced with C code can shed light on this?
